mkosi

```diff
@@ -457,6 +457,34 @@ def install_arch(args, workspace, run_build_script):
 
     subprocess.run(cmdline, check=True)
 
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-networkd"], check=True)
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-resolved"], check=True)
+    os.remove(os.path.join(workspace, "root", "etc/resolv.conf"))
+    os.symlink("../usr/lib/systemd/resolv.conf", os.path.join(workspace, "root", "etc/resolv.conf"))
+
+    nsswitch_old = os.path.join(workspace, "root", "etc/nsswitch.conf")
+    nsswitch_new = os.path.join(workspace, "root", "etc/nsswitch.conf.new")
+
+    with open(nsswitch_old, "r") as old:
+        with open(nsswitch_new, "w") as new:
+            for line in old:
+                if line.startswith("hosts:"):
+                    new.write(" ".join(["resolve" if x == "dns" else x for x in line.split(" ")]))
+                else:
+                    new.write(line)
+
+    shutil.copystat(nsswitch_old, nsswitch_new)
+    os.remove(nsswitch_old)
+    shutil.move(nsswitch_new, nsswitch_old)
+
+    if args.bootable:
+        with open(os.path.join(workspace, "root", "etc/systemd/network/wired.network"), "w") as f:
+            f.write("[Match]\n")
+            f.write("Name=en*\n")
+            f.write("\n")
+            f.write("[Network]\n")
+            f.write("DHCP=yes\n")
+
     print_step("Installing ArchLinux complete.")
```

Comment:
for now it would be OK to add this with the code you proposed as long as the systemd RFE is not implemented. But please use Type=ether as match, and make this a separate function, so that we can reuse this for the fedora case.